crates/polars-ops/src/chunked_array/strings/namespace.rs

```diff
@@ -357,6 +358,38 @@ pub trait Utf8NameSpaceImpl: AsUtf8 {
         Ok(out)
     }
 
+    /// Count all successive non-overlapping regex matches.
+    fn count_match_many(&self, pat: &Utf8Chunked) -> PolarsResult<UInt32Chunked> {
+        let ca = self.as_utf8();
+        polars_ensure!(
+            ca.len() == pat.len(),
+            ComputeError: "pattern's length: {} does not match that of the argument series: {}",
+            pat.len(), ca.len(),
+        );
+
+        // Very simple cache: don't recompile the same regex for multiple values in a row.
+        // TODO: proper LRU cache with reasonable memory limit.
+        let mut reg = Regex::new("").unwrap();
+        let mut lastpat = "";
+
+        let out: UInt32Chunked = ca
```

Comment:
Can you use `try_binary_elementwise` here. The `iterators` have a double indirection and therefore are slower. `try_binary_elementwise` should be less code and more performant.